core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java

```diff
@@ -105,6 +109,26 @@ public static Comparator<RewriteFileGroup> comparator(RewriteJobOrder rewriteJob
         return Comparator.comparing(RewriteFileGroup::numFiles);
       case FILES_DESC:
         return Comparator.comparing(RewriteFileGroup::numFiles, Comparator.reverseOrder());
+      case FILES_MIN_SEQUENCE_NUMBER_ASC:
+        return Comparator.comparing(RewriteFileGroup::minDataSequenceNumber);
+      case FILES_MIN_SEQUENCE_NUMBER_DESC:
+        return Comparator.comparing(
+            RewriteFileGroup::minDataSequenceNumber, Comparator.reverseOrder());
+      default:
+        return (fileGroupOne, fileGroupTwo) -> 0;
+    }
+  }
+
+  public static Comparator<FileScanTask> taskComparator(RewriteJobOrder rewriteJobOrder) {
+    switch (rewriteJobOrder) {
+      case FILES_ASC:
+        return Comparator.comparing(FileScanTask::length);
+      case FILES_DESC:
+        return Comparator.comparing(FileScanTask::length, Comparator.reverseOrder());
```

Comment:
```suggestion
      case BYTES_ASC:
        return Comparator.comparing(FileScanTask::length);
      case BYTES_DESC:
        return Comparator.comparing(FileScanTask::length, Comparator.reverseOrder());
```

FileScanTask::length matches expectation for BYTES_ASC and BYTES_DESC.

Reply:
I was not sure about the purpose of bytes ordering, rewriting smaller/larger file groups first? With that it's not clear how tasks sorting helps. For file count ordering I can see that it might be beneficial to rewrite smaller files first, or break down larger files first, and sorting by task length helps.